stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -4275,6 +4278,46 @@ impl NakamotoChainState {
             .as_ref()
             .map(|rewards| rewards.reward_info.clone());
 
+        if do_not_advance {
```

Comment:
Can this code block be its own method?  This function is already huge.

Reply:
Makes sense, updated in https://github.com/stacks-network/stacks-core/pull/5346/commits/84950edfa160e6662b3cbd41a14522fcf2fb9ba7